src/shogun/io/streaming/ParseBuffer.h

```diff
@@ -220,6 +225,18 @@ template <class T> class CParseBuffer: public CSGObject
 	bool free_vectors_on_destruct;
 };
 
+
+template <class T> void CParseBuffer<T>::init_vector()
+{
+	if (!free_vectors_on_destruct)
+		return;
+	for (int32_t i=0; i<ring_size; i++)
+	{
+		if(ex_ring[i].fv==NULL)
+			ex_ring[i].fv = new T();
```

Comment:
I tried `SG_MALLOC(T, 1)`, `SG_CALLOC(T, 1)`, `SG_MALLOC(T, sizeof(T) )`, and `SG_CALLOC(T, sizeof(T) )`; and SG_FREE is used. All these attempts are failed.
I have to use `new` and `delete`.